### DISCUSSION TITLE
`use mise` and `direnv` feedback

### DISCUSSION THREAD

**Discussion by wizzardich** [jdx/mise#2023] [General]
Hello!

I just noticed the note on `use mise` documentation [page](https://mise.jdx.dev/direnv.html) asking for feedback. I just wanted to pitch in: we do use it extensively and it seems to be working perfectly well. 

This allows us to have a workflow where we utilize the strong part of direnv (such as `source_if_exists` directives, etc), and use `mise` to provision tooling when entering the repository directory. I personally find this workflow both very intuitive and providing us with a nice way to "scope" tooling to the project.

**Comment by ssbarnea** [comment]
@wizzardich Can you please elaborate on how you use `source_if_exists` to activate mise? I am quite interested about the subject. Documentation page does not mention anything about it.

At this moment I ended up adding this to my .envrc file to make it work in a portable way:

```shell
type mise >/dev/null && {
  if [ ! -f ~/.config/direnv/lib/use_mise.sh ]; then
    echo "use_mise not found, create it first by running: mise direnv activate > ~/.config/direnv/lib/use_mise.sh"
    exit 2
  fi
  use mise
}
```

**Reply by wizzardich** [reply]
I'm afraid my usecase is nowhere near as complex. Because multiple different repository contributors may or may not have `mise` (for some - still `asdf`) installed, I just have this in my `.envrc`:

```
source_env_if_exists .ignored/.envrc
```

where `.ignored` is a dir that is ignored by git.

And code similar to yours resides in `.ignored/.envrc`. This way I can have mise manage my project, even if someone else utilizes `.tool-versions` with `asdf`. Plus, it's convenient for env vars that are temporary or shouldn't land in the repo.
